### ISSUE TITLE
Add new AA information to DASMDataset and calculate per-AA loss

### ISSUE THREAD

**Issue by matsen** [matsengrp/netam#65]
* [x] Add `aa_children_idxs` just like `aa_parents_idxs`
* [x] Calculate a categorical cross-entropy rather than BCE (but keep the BCE loss function, see below)
* [x] When we use the loss function from #63 (the existing BCE loss function) does this version look comparable?

If the performance is significantly worse, then that motivates keeping the existing loss function and just adding an additional loss function that will be "what amino acid is good conditioned on there being a substitution" to the BCE loss, and perhaps slowly ramping it up.
